test/unit/org/apache/cassandra/index/sai/cql/DropIndexWhileQueryingTest.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.plan.QueryController;
+import org.apache.cassandra.inject.ActionBuilder;
+import org.apache.cassandra.inject.Injection;
+import org.apache.cassandra.inject.Injections;
+
+import org.junit.Test;
+
+import static org.apache.cassandra.inject.InvokePointBuilder.newInvokePoint;
+import static org.junit.Assert.assertTrue;
+
+public class DropIndexWhileQueryingTest extends SAITester
+{
+    @Test
+    public void testReproductionForCNDB10732() throws Throwable
+    {
+        createTable("CREATE TABLE %s (electronicVehicle TEXT PRIMARY KEY, modelYear INT, make TEXT, vehType TEXT) WITH compaction = " +
+                "{'class' : 'SizeTieredCompactionStrategy', 'enabled' : false }");
+
+        createIndex("CREATE CUSTOM INDEX ON %s(make) USING 'StorageAttachedIndex'");
+        String indexName = createIndex("CREATE CUSTOM INDEX ON %s(modelYear) USING 'StorageAttachedIndex'");
+        createIndex("CREATE CUSTOM INDEX ON %s(vehType) USING 'StorageAttachedIndex'");
+        waitForTableIndexesQueryable();
+
+        Injection injection = Injections.newCustom("drop_index")
+                                        .add(newInvokePoint()
+                                             .onClass(QueryController.class)
+                                             .onMethod("buildPlan")
+                                             .atEntry())
+                                        .add(ActionBuilder
+                                             .newActionBuilder()
+                                             .actions()
+                                             .doAction("org.apache.cassandra.index.sai.cql.DropIndexWhileQueryingTest" +
+                                                       ".dropIndexForBytemanInjections" +
+                                                       "(\"DROP INDEX cql_test_keyspace." + indexName + "\")"))
+                                        .build();
+
+        Injections.inject(injection);
+        injection.enable();
+        assertTrue("Injection should be enabled", injection.isEnabled());
+        execute("INSERT INTO %s (electronicVehicle, modelYear, make, vehType) VALUES (?, ?, ?, ?)", "car", 1998, "Ford", "SUV");
+        assertInvalidMessage("An index may have been dropped.Cannot execute this query as it might involve " +
+                             "data filtering and thus may have unpredictable performance. If you want to execute this query" +
+                             " despite the performance unpredictability, use ALLOW FILTERING",
+                             "SELECT * FROM %s WHERE modelYear IN (1998, 2000) OR (make IN ('FORD', 'OPEL' ) OR vehType IN ('car', 'truck'));");
```

Comment:
It might be useful to repeat the query once again, to verify that it goes back to normal:
```
String query = "SELECT * FROM %s WHERE modelYear IN (1998, 2000) OR (make IN ('FORD', 'OPEL' ) OR vehType IN ('car', 'truck'))";
assertInvalidMessage(QueryController.INDEX_MAY_HAVE_BEEN_DROPPED, query);
assertInvalidMessage(StatementRestrictions.REQUIRES_ALLOW_FILTERING_MESSAGE, query);
```